mountpoint-s3-client/src/endpoint_config.rs

```diff
@@ -108,7 +113,7 @@ impl EndpointConfig {
     }
 
     /// resolve the endpoint from the [EndpointConfig] and the bucket name
-    pub fn resolve_for_bucket(&self, bucket: &str) -> Result<Uri, EndpointError> {
+    pub fn resolve_for_bucket(&self, bucket: &str) -> Result<ResolvedEndpoint, EndpointError> {
```

Comment:
We probably need to wrap `ResolvedEndpoint` (which is from `mountpoint-s3-crt`) in a new type defined here

Reply:
But, Uri is also from `mountpoint-s3-crt`?